src/main/java/frc/robot/subsystems/ControlPanelSubsystem.java

```diff
@@ -0,0 +1,79 @@
+package frc.robot.subsystems;
+
+import java.util.List;
+
+import com.revrobotics.ColorSensorV3;
+import com.revrobotics.ColorMatch;
+
+import com.ctre.phoenix.motorcontrol.can.VictorSPX;
+
+import edu.wpi.first.wpilibj.DoubleSolenoid;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj.I2C.Port;
+import edu.wpi.first.wpilibj.util.Color;
+
+public class ControlPanelSubsystem extends SubsystemBase {
+
+    public VictorSPX cpMotor;
+    public DoubleSolenoid cpSolenoid;
+    public ColorSensorV3 colorSensor;
+    public DriverStation ds;
+    public ColorMatch colorMatcher;
+
+    Color turnToColor = null;
+    public Color detectedColor = null;
+    public Color fmsColor = null;
+    Color lastColor = null;
+
+    List<String> colorNames;
+    public List<Color> colors;
+
+    //TODO: Both of these were "will_reset_to" variables in python, but I couldn't find a java alternative
```

Comment:
will_reset_to's were special in robotpy. You'll just have to manually reset them at the end of each...loop...

Reply:
I think the best way to do this is in the execute method.